firedrake/assemble.py

```diff
@@ -160,8 +162,10 @@ def _assemble(f, tensor=None, bcs=None, form_compiler_parameters=None,
         form_compiler_parameters = {}
     form_compiler_parameters["assemble_inverse"] = inverse
 
-    kernels = tsfc_interface.compile_form(f, "form", parameters=form_compiler_parameters,
-                                          inverse=inverse)
+    if isinstance(f, slate.Tensor):
+        kernels = slac.compile_slate_expression(f)
```

Comment:
This transformation will happen every time `assemble` is called on a slate expression, which is going to be an unacceptable performance hit.  I don't particularly care about disk caching, but can you do in memory caching in a similar manner to what happens in `tsfc_interface`?